### PR TITLE
When aggregating reports, inflate one expanded datatable at a time

### PR DESCRIPTION
**Pull request by diosmosis**
### Description:

Currently when aggregating reports over multi periods, we get the blob data for every child period, inflate the blob data for every period into a DataTable tree, then add them together. Afterwards we clean up the extra DataTables, but in the meantime a lot of memory is used, especially when the reports are large. This PR switches to inflating the DataTable tree for just one period at a time and should use less memory overall.

### Review

* [ ] Functional review done
* [ ] Potential edge cases thought about (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] Usability review done (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] Security review done [see checklist](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] Code review done
* [ ] Tests were added if useful/possible
* [ ] Reviewed for breaking changes
* [ ] Developer changelog updated if needed
* [ ] Documentation added if needed
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Comment by diosmosis** [comment]
Transitions screenshot change is not correct, but it is due to an unrelated bug: https://github.com/matomo-org/matomo/issues/17875

File: tests/PHPUnit/System/ImportLogsTest.php [hunk 1]
```diff
@@ -105,62 +105,6 @@ public function getApiForTesting()
         return $apis;
     }
 
-    /**
```
Comment (sgiehl) [inline tests/PHPUnit/System/ImportLogsTest.php]: Any reason for removing those tests?

File: tests/PHPUnit/System/ImportLogsTest.php [hunk 2]
```diff
@@ -78,7 +78,7 @@ public function getApiForTesting()
             // Mysqli rounds latitude/longitude
             $apiMethods = array('Live.getLastVisitsDetails');
         }
-        $apiMethods[] = 'Actions';
+        $apiMethods[] = 'Actions.getPageUrls';
```
Comment (sgiehl) [inline tests/PHPUnit/System/ImportLogsTest.php]: shouldn't there be quite  few file removals for the API methods that aren't called anymore, or wasn't this change on purpose?

File: plugins/Contents/tests/System/expected/test_Contents__Actions.getPageUrls_month.xml [hunk 3]
```diff
@@ -6,23 +6,23 @@
 		<nb_hits>2</nb_hits>
 		<sum_time_spent>540</sum_time_spent>
 		<nb_hits_with_time_network>2</nb_hits_with_time_network>
-		<min_time_network>0.0330</min_time_network>
-		<max_time_network>0.0330</max_time_network>
+		<min_time_network>0.033</min_time_network>
```
Comment (sgiehl) [inline plugins/Contents/tests/System/expected/test_Contents__Actions.getPageUrls_month.xml:9]: Is there a specific reason why the number formatting for those metrics changed?

**Review by sgiehl — CHANGES_REQUESTED** [review]

File: tests/PHPUnit/System/ImportLogsTest.php (hunk 1 above)
Comment (diosmosis) [reply tests/PHPUnit/System/ImportLogsTest.php]: That was for debugging, will undo.

File: plugins/Contents/tests/System/expected/test_Contents__Actions.getPageUrls_month.xml (hunk 3 above)
Comment (diosmosis) [reply plugins/Contents/tests/System/expected/test_Contents__Actions.getPageUrls_month.xml:9]: When querying, the data is a string '0.0330' in day reports. When aggregating, it is converted to a float, `0.033` and is stored that way in the datatable. This PR does not re-use rows in other tables when the row doesn't exist in the to be aggregated one, it uses `sumRow` on a `new Row()`, so the string value is converted to a float immediately.

**Comment by diosmosis** [comment]
@sgiehl removed the debugging code and fixed some tests

File: plugins/Contents/tests/System/expected/test_Contents__Actions.getPageUrls_month.xml (hunk 3 above)
Comment (sgiehl) [reply plugins/Contents/tests/System/expected/test_Contents__Actions.getPageUrls_month.xml:9]: that makes sense 👍

**Review by sgiehl — APPROVED** [review]
Looks good to me. Tests seem to be passing again 👍 
Not sure if maybe @tsteur wants to have another quick look before merging?

**Comment by tsteur** [comment]
Looks good to me @diosmosis 👍  Great, this will reduce memory quite a bit 💯

**Pull request merged**